gustaf/spline/microstructure/generator.py

```diff
@@ -0,0 +1,543 @@
+import numpy as np
+
+from gustaf._base import GustafBase
+from gustaf.spline import base
+
+
+class Generator(GustafBase):
+    """
+    Helper class to facilitatae the construction of microstructures
+    """
+
+    def __init__(
+            self,
+            deformation_function=None,
+            tiling=None,
+            microtile=None,
+            parametrization_function=None
+    ):
+        """
+        Helper class to facilitatae the construction of microstructures
+
+        Parameters
+        ----------
+        deformation_function : spline
+          Outer function that describes the contour of the microstructured
+          geometry
+        tiling : list of integers
+          microtiles per parametric dimension
+        microtile : spline or list of splines
+          Representation of the building block defined in the unit cube
+        parametrization_function : Callable (optional)
+          Function to describe spline parameters
+        """
+        if deformation_function is not None:
+            self.deformation_function = deformation_function
+
+        if tiling is not None:
+            self.tiling = tiling
+
+        if microtile is not None:
+            self.microtile = microtile
+
+        if parametrization_function is not None:
+            self.parametrization_function = parametrization_function
+
+    @property
+    def deformation_function(self):
```

Comment:
sounds a bit too foreign to me (calling spline a function), but I guess it is a standard term for this, correct?

Reply:
It is the notation that Gershon uses in his papers. It basically refers to the outer geometry. Generally I would like to stick to his notation but I am fine calling it MacroForm or OuterGeometry

Reply:
Im okay with this. Now that we talked about it, it is not foreign anymore